lib/bolt/inventory/group.rb

```
@@ -239,7 +239,16 @@ def validate_group_input(input)
           raise ValidationError.new(msg, @name)
         end
 
-        unless (unexpected_keys = input.keys - GROUP_KEYS).empty?
+        if input.key?('nodes')
+          msg = <<~MSG
+                Found deprecated key nodes in group #{@name}. Nodes keys are only available in v1 inventory
+                files, which are no longer supported by Bolt. Migrate to a v2 inventory file automatically
+                using 'bolt project migrate'.
+                MSG
+          @logger.warn(msg)
```

We should raise an exception here rather than just ignore the key.